extensions/wikia/Email/Controller/BlogPostController.class.php

```diff
@@ -0,0 +1,209 @@
+	 */
+	private function assertValidParams() {
+		$this->assertValidTitle();
+		$this->assertValidPostTitle();
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->pageTitle instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->pageTitle->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidPostTitle() {
+		if ( !$this->postTitle instanceof \Title ) {
+			$articleID = $this->getVal( 'childArticleID', false );
+			throw new Check( "Could not find post for article ID '$articleID' given by childArticleID" );
+		}
+
+		if ( !$this->postTitle->exists() ) {
+			throw new Check( "Post doesn't exist." );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'detailsHeader' => $this->getDetailsHeader(),
+			'details' => $this->getDetails(),
+			'buttonText' => $this->getPostLabel(),
+			'buttonLink' => $this->getPostLink(),
+			'contentFooterMessages' => [
+				$this->getPostListingLink(),
+			],
+			'hasContentFooterMessages' => true
+		] );
+	}
+
+	protected function getSummary() {
+		$blogURL = $this->pageTitle->getFullURL();
+		$listingTitle = $this->pageTitle->getText();
+
+		return wfMessage( $this->getSummaryKey(), $blogURL, $listingTitle )
+			->inLanguage( $this->targetLang )
+			->parse();
+	}
+
+	abstract protected function getSummaryKey();
+
+	protected function getDetails() {
+		$article = \Article::newFromTitle( $this->postTitle, \RequestContext::getMain() );
+		$service = new \ArticleService( $article );
+		$snippet = $service->getTextSnippet();
+
+		return $snippet;
+	}
+
+	protected function getDetailsHeader() {
+		$post = $this->postTitle;
+		return $post->getSubpageText();
+	}
+
+	protected function getPostLabel() {
+		return wfMessage( 'emailext-blogpost-link-label')
+			->inLanguage( $this->targetLang )
+			->parse();
+	}
+
+	protected function getPostLink() {
+		$post = $this->postTitle;
+		return $post->getFullURL();
+	}
+
+	protected function getFooterMessages() {
+		$parentUrl = $this->pageTitle->getFullURL( 'action=unwatch' );
+		$parentTitleText = $this->pageTitle->getPrefixedText();
+
+		$footerMessages = [
+			wfMessage( 'emailext-unfollow-text', $parentUrl, $parentTitleText )
+				->inLanguage( $this->targetLang )
+				->parse()
+		];
+		return array_merge( $footerMessages, parent::getFooterMessages() );
+	}
+
+	protected function getPostListingLink() {
+		$url = $this->pageTitle->getFullURL();
+		$name = $this->pageTitle->getText();
+
+		return wfMessage( 'emailext-blogpost-view-all', $url, $name )
```

Comment:
These should change to use $this->getMessage when/if? SOC-632 branch is merged into email_sprint_14 branch.